### PR TITLE
deploy-federation.sh: Wait for the propagation of clusterregistry CRD

### PR DESCRIPTION
**Pull request by yamt**
To workaround the following error on the following invocation of
"kubefed2 enable".

customresourcedefinition.apiextensions.k8s.io/clusters.clusterregistry.k8s.io created
F0304 09:24:34.796992    2281 enable.go:111] error: Error listing api resources: unable to retrieve the complete list of server APIs: clusterregistry.k8s.io/v1alpha1: the server could not find the requested resource

### CODE REVIEW

**Comment by yamt** [comment]
the ci failure should be https://github.com/kubernetes-sigs/federation-v2/pull/620

File: scripts/deploy-federation.sh [hunk 1]
```diff
@@ -67,6 +67,11 @@ function deploy-with-script() {
   # TODO(marun) Ensure federatdtypeconfig is available before creating instances
   # TODO(marun) Ensure crds are created for a given federated type before starting sync controller for that type
 
+  # Wait for the propagation of the clusterregistry CRD
+  while ! kubectl api-resources | grep clusterregistry.k8s.io > /dev/null; do
+    sleep 1
+  done
+
```
Comment (xunpan) [inline scripts/deploy-federation.sh]: we had an function for such kind of wait condition check `util::wait-for-condition`. Please try to use it.

Comment (yamt) [reply scripts/deploy-federation.sh]: thank you. i wasn't aware of it. i'll try.

Comment (yamt) [reply scripts/deploy-federation.sh]: done

**Comment by marun** [comment]
Thank you!

/approve

**Pull request merged**